lib/zip/file.rb

```diff
@@ -51,21 +51,31 @@ class File < CentralDirectory
     DATA_BUFFER_SIZE     = 8192
     IO_METHODS           = [:tell, :seek, :read, :close]
 
+    DEFAULT_OPTIONS = {
+      restore_ownership:   false,
+      restore_permissions: false,
```

Comment:
It looks like this used to default to `true` (and, unlike `restore_times`, did actually do something). Is there a particular reason to default it to `false` now?

Reply:
It did default to `true`, but it didn't do anything! I forgot to mention this in the original PR.

In `Entry#create_file` the call to set the attributes of a file (`set_extra_attributes_on_path`) was being called too early - before the file was closed - so the permissions weren't being set correctly. In the course of my PR I moved the call to `set_extra_attributes_on_path` to when the file is closed to fix this. (`Entry#create_directory` got this right already.)

Then I set this to be be `false` by default for consistency. I certainly think that permissions *should* be restored by default though.

I'm happy to change this to `true` while I'm making the other changes required.

Reply:
Aha, I didn't pick up on that. 👍 for `false` in that case.

Reply:
OK, I will leave it set to false for now then.